src/python/gudhi/tensorflow/cubical_layer.py

```diff
@@ -0,0 +1,72 @@
+import numpy               as np
+import tensorflow          as tf
+from ..cubical_complex  import CubicalComplex
+
+######################
+# Cubical filtration #
+######################
+
+# The parameters of the model are the pixel values.
+
+def _Cubical(X, dimensions):
+    # Parameters: X (image),
+    #             dimensions (homology dimensions)
+
+    # Compute the persistence pairs with Gudhi
+    Xs = X.shape
+    cc = CubicalComplex(dimensions=Xs, top_dimensional_cells=X.flatten())
```

Comment:
I think you don't need to flatten and pass explicit dimensions anymore, it should just do the right thing with a multidimensional numpy array.

Reply:
Thanks! Done.